src/manifest.rs

```diff
@@ -109,94 +155,200 @@ impl Target {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::fs;
+    use std::path;
+    use std::process;
 
-    fn read_manifest(filename: &str) -> Manifest {
-        let manifest_str: String =
-            fs::read_to_string(filename).expect("manifest file cannot be read");
+    fn read_manifest(directory: &path::Path) -> Manifest {
+        let output = process::Command::new("cargo")
```

Comment:
With this change we would be running `cargo` as a process within our unit tests.  What are the benefits of doing this instead of using a pre-generated manifest?

Reply:
Several reasons: 
- I needed to change all the manifests because cargo metadata has a slightly different output.
- Having some example projects is handy since we can reuse them for other tests (Integration tests)
- Given the added complexity of workspace I feared the pre-generated ones might become non representative to quick and fool us with false positives
- I think we should replace this part with [cargo-metadata](https://crates.io/crates/cargo_metadata) anyway so I spent minimal time on this (I needed the example projects to verify some assumptions)

Reply:
I thought about making a separate script to pre generate them and commit them to have best of both, but figured this was easier